### ISSUE TITLE
Add clarifying tooltips to header navigation

### ISSUE THREAD

**Issue by KrooshalUX** [opensearch-project/OpenSearch-Dashboards#3573]
**Is your feature request related to a problem? Please describe.**

Related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3557 , proper tool tips do not appear on hover for Menu or Help icons. 

**Describe the solution you'd like**

Similar to the tool tip that appears on "Home" icon in the stacked nav or on the OpenSearch logo icon in the single nav, add tool tips that describe, in the simplest terms, what the icons represent. 

**Describe alternatives you've considered**

In order to further disambiguate the purpose of these icons, the UX team is currently working on redesigning our icon offering. 
Example: https://github.com/opensearch-project/ux/issues/11

However, regardless of the direction that an icon redesign may take, proper tool tips are a must-have. 

**Additional context**
<img width="1792" alt="Screen Shot 2023-03-09 at 10 40 00 AM" src="https://user-images.githubusercontent.com/105819041/224123605-a14f830d-8596-4eb5-8314-8e73d5da2494.png">

**Comment by KrooshalUX** [comment]
Tool tip copy request can be found here: https://github.com/opensearch-project/documentation-website/issues/3351

**Comment by joshuarrrr** [comment]
As a note for developers, the tooltip text can be set by specifying the `title` prop of the `EuiIcon` component. See https://oui.opensearch.org/1.0/#/display/icons for more details.

**Comment by sayuree** [comment]
I added a tooltip to Help icon, in the [previous discussion](https://github.com/opensearch-project/documentation-website/issues/3351#issuecomment-1466379362) the conclusion was the following:
`I agree with Fanit. "Help" and "Home". No tooltip necessary for the menu icon.`
'Home' icon has a tooltip.

**Comment by KrooshalUX** [comment]
> I added a tooltip to Help icon, in the [previous discussion](https://github.com/opensearch-project/documentation-website/issues/3351#issuecomment-1466379362) the conclusion was the following: `I agree with Fanit. "Help" and "Home". No tooltip necessary for the menu icon.` 'Home' icon has a tooltip.

@sayuree I did not yet provide the final UX direction for this issue in the comments here as I was finalizing my discussions, apologies I should have tagged this issue with "blocked" until I had time to update it. 

Regardless of what the docs team had provided, for accessibility purposes, the Menu icon _does in fact need the tool tip_. So, please adjust the PR that you have as well. Thanks!
